.circleci/config.yml

```diff
@@ -546,12 +468,45 @@ jobs:
           study_key: fon
 
   app-build-and-store-job:
+    description: Build and store build artifacts to Google cloud storage
     working_directory: *ng_workspace_path
     executor:
       name: build-executor
+    parameters:
+      study_key:
+        type: string
     steps:
-      - app-build-and-store:
-          study_key: << pipeline.parameters.study_key >>
+      - build:
+          study_key: << parameters.study_key >>
+      - run:
+          name: Don't archive build artifacts if branch is a pull request
+          command: |
+            if [[ ${CIRCLE_PULL_REQUEST} ]]; then
+              echo "CIRCLE_PULL_REQUEST: $CIRCLE_PULL_REQUEST"
+              echo "CIRCLE_BRANCH: $CIRCLE_BRANCH"
+              echo "Skip archive build"
+              circleci-agent step halt
+            fi
```

Comment:
This is a new check condition to halt job if job was triggered by pull requests. Because we don't want to store pull requests' build to GSA.

Reply:
what is GSA?